### PR TITLE
WIP: ref(*): Use confd 0.7

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Upgrade confd from 0.5 to 0.7 and converted all templates to the new syntax. 

This is **not completed yet**, but I opened this already for early feedback and to start a small discussion how to convert some template patterns.

Requirement for https://github.com/deis/deis/issues/2160

### CODE REVIEW

File: controller/templates/confd_settings.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # security keys and auth tokens
-SECRET_KEY = '{{ .deis_controller_secretKey }}'
-BUILDER_KEY = '{{ .deis_controller_builderKey }}'
+SECRET_KEY = '{{ getv "/deis/controller/secretKey" }}'
+BUILDER_KEY = '{{ getv "/deis/controller/builderKey" }}'
 
 # scheduler settings
 SCHEDULER_MODULE = '{{ or (.deis_controller_schedulerModule) "fleet" }}'
```
Comment (johanneswuerbach) [inline controller/templates/confd_settings.py]: I couldn't find a nice way to express default values in confd 0.7

0.5
`{{ or (.deis_controller_schedulerModule) "fleet" }}`
0.7
`{{ if exists "/deis/controller/schedulerModule" }}{{ getv "/deis/controller/schedulerModule" }}{{ else }}fleet{{ end }}`

Deis uses default values a lot, somebody has a better idea?

**Comment by aledbf** [comment]
@johanneswuerbach  :+1:

**Comment by mboersma** [comment]
+1 for updating confd. I don't know the new template idioms yet but I'll try to help out here if I can.

**Comment by mboersma** [comment]
It would be outstanding if this could also resolve #2299 (Log which keys are missing).

**Comment by johanneswuerbach** [comment]
Yep, but it's currently blocked on not having a default value pattern in confd 0.7 https://github.com/kelseyhightower/memkv/pull/4

**Comment by bacongobbler** [comment]
@johanneswuerbach text/template has a native `or` function. I'm one of the maintainers of confd now so I'll see if I can get some good examples going.

http://golang.org/pkg/text/template/#hdr-Functions

**Comment by bacongobbler** [comment]
sadly I couldn't find a better alternative than `if exists/else` in the new confd since the memkv functions all return errors now instead of `<no value>`.

**Comment by johanneswuerbach** [comment]
Yes, I tried to add another one, but it isn't merged yet. https://github.com/kelseyhightower/memkv/pull/4 definitely open for other solutions.

**Comment by bacongobbler** [comment]
We're gonna have to address the router later because confd 0.8 is *practically* impossible to work with for our use case, so we're gonna have to stick with a very old version of confd for that component. As for all the others, I've filed #3360 #3360 #3364 #3365 and #3366 in separate PRs so each component can get the love it deserves.

Thanks for spear-heading this, @johanneswuerbach :)

**Comment by johanneswuerbach** [comment]
Thanks, those PRs are looking great. The router template was also the biggest barrier for me :-D

**Pull request closed without merging**